src/common/robo_claw.cc

```diff
@@ -0,0 +1,136 @@
+    const uint8_t byte = (uint8_t)std::round(127.0 * std::min(1.0f, forward ? throttle : -throttle));
+
+    writeCommand(forward ? Command::M2FORWARD : Command::M2BACKWARD, byte);
+}
+//----------------------------------------------------------------------------
+uint32_t RoboClaw::getMotor1Encoder()
+{
+    uint32_t encoder = 0;
+    uint8_t status = 0;
+    readCommand(Command::GETM1ENC, &encoder, &status);
+
+    // **TODO** think about something sensible to do with status
+    return encoder;
+}
+//----------------------------------------------------------------------------
+uint32_t RoboClaw::getMotor2Encoder()
+{
+    uint32_t encoder = 0;
+    uint8_t status = 0;
+    readCommand(Command::GETM2ENC, &encoder, &status);
+
+    // **TODO** think about something sensible to do with status
+    return encoder;
+}
+//----------------------------------------------------------------------------
+void RoboClaw::resetEncoders()
+{
+    writeCommand(Command::RESETENC);
+}
+//----------------------------------------------------------------------------
+int RoboClaw::getMotor1Speed()
+{
+    uint32_t speed = 0;
+    uint8_t status = 0;
+    readCommand(Command::GETM1SPEED, &speed, &status);
+
+    // **NOTE** status signifies direction but this also seems to be 2s compliment encoded in speed
+    return (int)speed;
+}
+//----------------------------------------------------------------------------
+int RoboClaw::getMotor2Speed()
+{
+    uint32_t speed = 0;
+    uint8_t status = 0;
+    readCommand(Command::GETM2SPEED, &speed, &status);
+
+    // Cast to signed int and return
+    // **NOTE** status signifies direction but this also seems to be 2s compliment encoded in speed
+    return (int)speed;
+}
+//----------------------------------------------------------------------------
+std::string RoboClaw::getVersion()
+{
+    // Retry
+    for(int r = 0; r < m_MaxRetry; r++) {
+        // Write address and command
+        CRC crc;
+        write(crc, m_Address, static_cast<uint8_t>(Command::GETVERSION));
+
+        // Read maximum version characters
+        uint8_t version[48];
+        for(uint8_t &c : version) {
+            m_SerialInterface.readByte(c);
+            crc.update(c);
+
+            // If end of string is reached
+            if(c == '\0') {
+                // Check CRC against one returned by device
+                if(!crc.check(m_SerialInterface)) {
+                    throw std::runtime_error("CRC check failed");
+                }
+
+                // Return string
+                return reinterpret_cast<char*>(version);
+            }
+        }
+
+        throw std::runtime_error("Version string not correctly terminated");
+    }
+
+    throw std::runtime_error("Unable to read to serial port");
+}
+//----------------------------------------------------------------------------
+float RoboClaw::getBatteryVoltage()
+{
+    uint16_t voltage;
+    readCommand(Command::GETMBATT, &voltage);
+    return (voltage * 0.1f);
```

Comment:
There is the ``volt_t`` type, but that's probably not necessary

Reply:
also those units are disabled by default so it would be kinda awkward